docs/adr/0005-network-device-name.md

```diff
@@ -0,0 +1,49 @@
+# 4. Merge Create and Update Plan into a Single Plan
+
+* Status: pending   // will be updated after PR review
+* Date: 2021-11-08
+* Authors: @yitsushi
+* Deciders: @Callisto13 @jmickey @richardcase @yitsushi
+
+## Context
+
+With the POC, network interface names are generated from the name of the
+MicroVM and its Namespace. It works for short VM name and Namespace
+combinations, but device names can't be longer than 15 bytes[^1][^2].
+
+Because of this limitation, we have to find a better way to generate network
+device name on the host.
+
+Considered options:
+
+* Generate a UUID and use the first N bytes.
+* Generate a hash of the Name and Namespace combination and use the first N
+  bytes.
+* Generate a random value.
+
+[^1]: https://elixir.bootlin.com/linux/v5.6/source/include/linux/netdevice.h#L1826
+[^2]: https://elixir.bootlin.com/linux/v5.6/source/include/uapi/linux/if.h#L33
+
+## Decision
+
+Following the device name generator in docker-ce[^3], we decided to use a random
+value. Docker-ce tries to generate a name 3 times, if the generated name is
+already taken. To reduce possible failures, we decided to retry 5 times,
+it's still not slow, but gives more opportunities on machines with a lot of
+network devices.
+
+[^3]: https://github.com/docker/docker-ce/blob/1093a93b336461032352e776893afefc2cf3a50d/components/engine/libnetwork/netutils/utils.go#L126
+
+## Consequences
+
+It is not possible to determine which MicroVM is the owner of the network
```

Comment:
i am sure you have already considered this, but is there no way to save the device name on the mvm state?

Reply:
We save it in the Status field of the Spec, but from the device name (`/dev/something` or `ip link show`) alone we don't know which vm.

Reply:
That's not entirely true, but I would say it's a huge hack and can be used for debugging or in the cleanup:
```
❯ cat /proc/$(sudo lsof /dev/tap$(cat /sys/class/net/flvtap6396803/ifindex) 2>/dev/null | awk '/firecrack/ { print $2 }')/cmdline
firecracker--api-sock/var/lib/flintlock/vm/ns1/mvm1/firecracker.sock--idns1-mvm1--boot-timer
```

Reply:
> And it can be used only if the firecracker instance is running.

ah yeh, true

Reply:
We could also add a utility command (a bit like the cleanup command we are going to add) to dump the vmspec, network interface name etc